common/flatpak-dir.c

```diff
@@ -1853,69 +1867,181 @@ repo_pull_one_dir (OstreeRepo          *self,
+                                      &collection_id, NULL))
+    g_clear_pointer (&collection_id, g_free);
 
-  if (dirs_to_pull)
+  if (collection_id != NULL)
     {
-      g_variant_builder_add (&builder, "{s@v}", "subdirs",
-                             g_variant_new_variant (g_variant_new_strv ((const char * const *)dirs_to_pull, -1)));
-      force_disable_deltas = TRUE;
+      GVariantBuilder find_builder, pull_builder;
+      g_autoptr(GVariant) find_options = NULL, pull_options = NULL;
+      g_autoptr(GMainContext) context = NULL;
+      g_autoptr(GAsyncResult) find_result = NULL, pull_result = NULL;
+      g_auto(OstreeRepoFinderResultv) results = NULL;
+      OstreeCollectionRef collection_ref;
+      OstreeCollectionRef *collection_refs_to_fetch[2];
+
+      g_variant_builder_init (&find_builder, G_VARIANT_TYPE ("a{sv}"));
+      g_variant_builder_init (&pull_builder, G_VARIANT_TYPE ("a{sv}"));
+
+      if (dirs_to_pull)
+        {
+          g_variant_builder_add (&pull_builder, "{s@v}", "subdirs",
+                                 g_variant_new_variant (g_variant_new_strv ((const char * const *)dirs_to_pull, -1)));
+          force_disable_deltas = TRUE;
+        }
+
+      if (force_disable_deltas)
+        {
+          g_variant_builder_add (&find_builder, "{s@v}", "disable-static-deltas",
+                                 g_variant_new_variant (g_variant_new_boolean (TRUE)));
+          g_variant_builder_add (&pull_builder, "{s@v}", "disable-static-deltas",
+                                 g_variant_new_variant (g_variant_new_boolean (TRUE)));
+        }
+
+      g_variant_builder_add (&pull_builder, "{s@v}", "inherit-transaction",
+                             g_variant_new_variant (g_variant_new_boolean (TRUE)));
+
+      g_variant_builder_add (&pull_builder, "{s@v}", "flags",
+                             g_variant_new_variant (g_variant_new_int32 (flags)));
+
+      collection_ref.collection_id = collection_id;
+      collection_ref.ref_name = (char *) ref_to_fetch;
+
+      collection_refs_to_fetch[0] = &collection_ref;
+      collection_refs_to_fetch[1] = NULL;
+
+      revs_to_fetch[0] = rev_to_fetch;
+      revs_to_fetch[1] = NULL;
+      g_variant_builder_add (&find_builder, "{s@v}", "override-commit-ids",
+                             g_variant_new_variant (g_variant_new_strv ((const char * const *) revs_to_fetch, -1)));
+
+      if (progress != NULL)
+        update_freq = GPOINTER_TO_UINT (g_object_get_data (G_OBJECT (progress), "update-frequency"));
+      if (update_freq == 0)
+        update_freq = FLATPAK_DEFAULT_UPDATE_FREQUENCY;
+
+      g_variant_builder_add (&find_builder, "{s@v}", "update-frequency",
+                             g_variant_new_variant (g_variant_new_uint32 (update_freq)));
+      g_variant_builder_add (&pull_builder, "{s@v}", "update-frequency",
+                             g_variant_new_variant (g_variant_new_uint32 (update_freq)));
```

Comment:
(Fixed in https://github.com/flatpak/flatpak/pull/884/commits/4bf896e971aa8cc82489d642eba0886c84498f24.)